src/test/java/com/payfurl/payfurlsdk/TestConfigProvider.java

```diff
@@ -1,23 +1,64 @@
 package com.payfurl.payfurlsdk;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.payfurl.payfurlsdk.http.client.config.Environment;
 import org.apache.commons.lang3.EnumUtils;
 import org.apache.commons.lang3.StringUtils;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 public final class TestConfigProvider {
-    private static final String LOCAL_ACCESS_SECRET_KEY = "SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c";
-    private static final String PAYFURL_ACCESS_TEST_SECRET_KEY = "PAYFURL_LOCAL_ACCESS_SECRET_KEY";
-    private static final String PAYFURL_ENVIRONMENT_KEY = "PAYFURL_ENVIRONMENT";
+    private static final String TEST_DATA_FILE = "appsettings.json";
+    private static final Map<String, Object> config = new HashMap<>();
+    private static int numToken = 0;
+
+    static {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            InputStream inputStream = TestConfigProvider.class.getClassLoader().getResourceAsStream(TEST_DATA_FILE);
+            config.putAll(mapper.readValue(inputStream, Map.class));
+        } catch (IOException e) {
+            throw new RuntimeException();
+        }
+    }
 
     private TestConfigProvider() {
         throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
     }
 
     public static String getSecretKeyWithFallback() {
-        return StringUtils.defaultIfEmpty(System.getProperty(PAYFURL_ACCESS_TEST_SECRET_KEY), LOCAL_ACCESS_SECRET_KEY);
+        return StringUtils.defaultIfEmpty(config.get("SecretKey").toString(), "");
     }
 
     public static Environment getEnvironmentWithFallback() {
-        return EnumUtils.getEnumIgnoreCase(Environment.class, System.getProperty(PAYFURL_ENVIRONMENT_KEY), Environment.LOCAL);
+        String env = config.get("Environment").toString();
```

Comment:
Let's extract `"Environment"` into some kind of test constants. For example, `TestConfigKeys`. Applicable to all cases where these types of keys are used: `"ProviderId"` etc.